src/main/java/org/checkerframework/checker/androidresource/AndroidResourceVisitor.java

```diff
@@ -0,0 +1,80 @@
+        AnnotatedTypeMirror rightOperandType = atypeFactory.getAnnotatedType(rightOperand);
+
+        Kind kind = node.getKind();
+
+//        System.out.println(node.toString() + ", " + node.getKind() + "\n");
+
+        if (checkOperandsHasResAnnotations(leftOperandType) | checkOperandsHasResAnnotations(rightOperandType)) {
+            checker.report(Result.warning("binary.operation.not.allowed", kind), node);
+        }
+
+        return super.visitBinary(node, aVoid);
+    }
+
+    /**
+     * Gives a warning whenever expression/variable in a compound assignment have a annotation of type @XXXRes.
+     */
+    @Override
+    public Void visitCompoundAssignment(CompoundAssignmentTree node, Void p) {
+
+        ExpressionTree expression = node.getExpression();
+        ExpressionTree variable = node.getVariable();
+        AnnotatedTypeMirror expressionType = atypeFactory.getAnnotatedType(expression);
+        AnnotatedTypeMirror variableType = atypeFactory.getAnnotatedType(variable);
+
+        Kind kind = node.getKind();
+
+//        System.out.println(node.toString() + ", " + node.getKind() + "\n");
+
+        if (checkOperandsHasResAnnotations(expressionType) | checkOperandsHasResAnnotations(variableType)) {
+            checker.report(Result.warning("compound.assignment.not.allowed", kind), node);
+        }
+
+        return super.visitCompoundAssignment(node, p);
+    }
+
+    /**
+     *  Method to check XXXRes annotations.
+     *
+     * @param annotatedTypeMirror
+     * @return whether the given [AnnotatedTypeMirror] contains the XXXRes annoations or not.
+     */
+    public boolean checkOperandsHasResAnnotations(AnnotatedTypeMirror annotatedTypeMirror) {
+        return annotatedTypeMirror.hasAnnotation(AnimatorRes.class) | annotatedTypeMirror.hasAnnotation(AnimRes.class)
+                | annotatedTypeMirror.hasAnnotation(AnyRes.class) | annotatedTypeMirror.hasAnnotation(ArrayRes.class)
+                | annotatedTypeMirror.hasAnnotation(AttrRes.class) | annotatedTypeMirror.hasAnnotation(BoolRes.class)
+                | annotatedTypeMirror.hasAnnotation(ColorRes.class) | annotatedTypeMirror.hasAnnotation(DimenRes.class)
+                | annotatedTypeMirror.hasAnnotation(DrawableRes.class) | annotatedTypeMirror.hasAnnotation(FractionRes.class)
+                | annotatedTypeMirror.hasAnnotation(IdRes.class) | annotatedTypeMirror.hasAnnotation(IntegerRes.class)
+                | annotatedTypeMirror.hasAnnotation(InterpolatorRes.class) | annotatedTypeMirror.hasAnnotation(LayoutRes.class)
+                | annotatedTypeMirror.hasAnnotation(MenuRes.class) | annotatedTypeMirror.hasAnnotation(PluralsRes.class)
+                | annotatedTypeMirror.hasAnnotation(RawRes.class) | annotatedTypeMirror.hasAnnotation(StringRes.class)
+                | annotatedTypeMirror.hasAnnotation(StyleableRes.class) | annotatedTypeMirror.hasAnnotation(XmlRes.class);

```

Comment:
Can you simplify this test by checking for the few annotations that are allowed in binary/composite operations?

Reply:
@wmdietl, I have some changes, please check.